src/password-confirmation.ts

```diff
@@ -54,13 +54,26 @@ async function _confirmPassword(password: string) {
 	logger.debug('Password confirmed')
 }
 
+// Internal state to track the dialog
+let _passwordDialog: Promise<boolean> | undefined
+let _dialogCallback: (s: string) => Promise<void>
+
 /**
  * Spawn a dialog to prompt the password.
  *
  * @param validate Is called to validate the user's password
  */
 async function promptPassword(validate: (password: string) => Promise<void>) {
-	const result = await spawnDialog(PasswordDialogVue, { validate })
+	_dialogCallback = validate
```

Comment:
```suggestion
	// prevent race condition: if a dialog is already open, reuse it
	if (_passwordDialog) {
		return _passwordDialog
	}
	_dialogCallback = validate
```

Reply:
No the validation callback needs to be set.

It will not be reused by another call from the user or app, but only from another call from the axios interceptors.

Request password -> send request with header -> invalid -> reuse dialog